### PR TITLE
Jupyter Notebook Estimator Plotting Utility

### LINKED ISSUE #216: Jupyter notebook estimator plotting and testing.

### PR DESCRIPTION
**Pull request by nhz2**
# Jupyter Notebook Estimator Plotting Utility

Fixes #216 

Depends on #226 

### Summary of changes
Added a Jupyter Notebook animated plotting function.
The function is inspired by testing estimators in Simulink, which has scopes to look at the performance of an estimator while a sim is running. The function can plot a list of functions of the estimator, sensors, and truth at a specified sample rate. 

The goal is to quickly test and see the accuracy of an estimator in Jupyter Notebook.
[psim/estimatortest/Orbit-estimator-test.ipynb](https://github.com/pathfinder-for-autonomous-navigation/psim/blob/Jupyter-Notebook-Plotting-Utility/estimatortest/Orbit-estimator-test.ipynb) is an example.

### Running the Jupyter Notebook on the linux server over ssh
1. ssh into the server
2. clone the psim repo add `--recursive` flag to also get lin
3. `cd psim`
4. `git checkout Jupyter-Notebook-Plotting-Utility`
5. `python -m venv venv`
6. `source venv/bin/activate`
7. `pip install -r requirements.txt`
8. `ipython kernel install --user --name=venv`
9. `jupyter notebook --no-browser --port 1234`
10. Start an SSH tunnel to the server, see [more info](https://medium.com/@apbetahouse45/how-to-run-jupyter-notebooks-on-remote-server-part-1-ssh-a2be0232c533)
on unix use `ssh -NL 1234:localhost:1234 username@serverip` in a new local terminal
11. copy the URL in the output from step 9 into your browser.
Something like the following should come up.
Screenshot 2020-05-08 17 15 54

12. Click `estimatortest` and then open `Orbit-estimator-test.ipynb`

13. Change the kernel to venv. 
Screenshot 2020-05-08 17 22 39

14. Everything should be working now. I would recommend opening another ssh connection to do git stuff, or make edits in vim.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -32,3 +32,8 @@ jobs:
       run: |
         python -m platformio run --verbose -e teensy35_ci
         python -m platformio run --verbose -e teensy36_ci
+
+    - name: Python Wrapper tests
+      run: |
+        pip install pytest
+        pytest estimatortest/
```
Comment (kylekrol) [inline .github/workflows/main.yml]: Just a heads up I think this may end up conflicting with the other PR.

Comment (kylekrol) [reply .github/workflows/main.yml]: Nevermind, I see just everything from the other PR is in here as well.

**Review by kylekrol — APPROVED** [review]
I was able to get everything up and running on my machine locally! I'm a little unsure as to why but I'd suggest trying to run the Jupyter notebook with explicitly installing the extra kernel (`ipython kernel install --user --name=venv`). This wasn't required on my machine - not sure why.

Also, you may want to add the following to the instructions so hd5 files get checked out:

    git lfs install         #< this one gets run once
    git lfs fetch
    git lfs checkout

Other than that, the plotter functions are sort of difficult to read through but that's not too terribly important. Looks good otherwise!

**Comment by tanishqaggarwal** [comment]
Is this good to merge?

**Pull request merged**